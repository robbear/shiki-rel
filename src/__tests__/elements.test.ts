import { getHighlighter, renderToHtml } from '../index'

test('Customize elements - pre', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['js']
  })
  const tokens = highlighter.codeToThemedTokens(`console.log('shiki');`, 'js')
  const out = renderToHtml(tokens, {
    fg: highlighter.getForegroundColor('nord'),
    bg: highlighter.getBackgroundColor('nord'),
    elements: {
      pre({ className, style, children }) {
        return `<pre tabindex="1" class="${className}" style="${style}">${children}</pre>`
      }
    }
  })
  expect(out).toMatchSnapshot()
})

test('Customize elements - code', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['js']
  })
  const tokens = highlighter.codeToThemedTokens(`console.log('shiki');`, 'js')
  const out = renderToHtml(tokens, {
    fg: highlighter.getForegroundColor('nord'),
    bg: highlighter.getBackgroundColor('nord'),
    elements: {
      code({ children }) {
        return `<code data-copy-text="hello">${children}</code>`
      }
    }
  })
  expect(out).toMatchSnapshot()
})

test('Customize elements - line', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['js']
  })
  const tokens = highlighter.codeToThemedTokens(`console.log('shiki');`, 'js')
  const out = renderToHtml(tokens, {
    fg: highlighter.getForegroundColor('nord'),
    bg: highlighter.getBackgroundColor('nord'),
    elements: {
      line({ className, index, children }) {
        return `<span class="${className}"><span class="line-no">${index}</span>${children}</span>`
      }
    }
  })
  expect(out).toMatchSnapshot()
})

test('Customize elements - token', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['js']
  })
  const tokens = highlighter.codeToThemedTokens(`console.log('shiki');`, 'js')
  const out = renderToHtml(tokens, {
    fg: highlighter.getForegroundColor('nord'),
    bg: highlighter.getBackgroundColor('nord'),
    elements: {
      token({ style, token, children }) {
        return `<span data-color="${token.color || ''}" style="${style}">${children}</span>`
      }
    }
  })
  expect(out).toMatchSnapshot()
})
