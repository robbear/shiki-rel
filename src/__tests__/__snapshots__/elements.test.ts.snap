// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customize elements - code 1`] = `"<pre class=\\"shiki\\" style=\\"background-color: #2e3440ff\\"><code data-copy-text=\\"hello\\"><span class=\\"line\\"><span style=\\"color: #D8DEE9\\">console</span><span style=\\"color: #ECEFF4\\">.</span><span style=\\"color: #88C0D0\\">log</span><span style=\\"color: #D8DEE9FF\\">(</span><span style=\\"color: #ECEFF4\\">&#39;</span><span style=\\"color: #A3BE8C\\">shiki</span><span style=\\"color: #ECEFF4\\">&#39;</span><span style=\\"color: #D8DEE9FF\\">)</span><span style=\\"color: #81A1C1\\">;</span></span></code></pre>"`;

exports[`Customize elements - line 1`] = `"<pre class=\\"shiki\\" style=\\"background-color: #2e3440ff\\"><code><span class=\\"line\\"><span class=\\"line-no\\">0</span><span style=\\"color: #D8DEE9\\">console</span><span style=\\"color: #ECEFF4\\">.</span><span style=\\"color: #88C0D0\\">log</span><span style=\\"color: #D8DEE9FF\\">(</span><span style=\\"color: #ECEFF4\\">&#39;</span><span style=\\"color: #A3BE8C\\">shiki</span><span style=\\"color: #ECEFF4\\">&#39;</span><span style=\\"color: #D8DEE9FF\\">)</span><span style=\\"color: #81A1C1\\">;</span></span></code></pre>"`;

exports[`Customize elements - pre 1`] = `"<pre tabindex=\\"1\\" class=\\"shiki\\" style=\\"background-color: #2e3440ff\\"><code><span class=\\"line\\"><span style=\\"color: #D8DEE9\\">console</span><span style=\\"color: #ECEFF4\\">.</span><span style=\\"color: #88C0D0\\">log</span><span style=\\"color: #D8DEE9FF\\">(</span><span style=\\"color: #ECEFF4\\">&#39;</span><span style=\\"color: #A3BE8C\\">shiki</span><span style=\\"color: #ECEFF4\\">&#39;</span><span style=\\"color: #D8DEE9FF\\">)</span><span style=\\"color: #81A1C1\\">;</span></span></code></pre>"`;

exports[`Customize elements - token 1`] = `"<pre class=\\"shiki\\" style=\\"background-color: #2e3440ff\\"><code><span class=\\"line\\"><span data-color=\\"#D8DEE9\\" style=\\"color: #D8DEE9\\">console</span><span data-color=\\"#ECEFF4\\" style=\\"color: #ECEFF4\\">.</span><span data-color=\\"#88C0D0\\" style=\\"color: #88C0D0\\">log</span><span data-color=\\"#D8DEE9FF\\" style=\\"color: #D8DEE9FF\\">(</span><span data-color=\\"#ECEFF4\\" style=\\"color: #ECEFF4\\">&#39;</span><span data-color=\\"#A3BE8C\\" style=\\"color: #A3BE8C\\">shiki</span><span data-color=\\"#ECEFF4\\" style=\\"color: #ECEFF4\\">&#39;</span><span data-color=\\"#D8DEE9FF\\" style=\\"color: #D8DEE9FF\\">)</span><span data-color=\\"#81A1C1\\" style=\\"color: #81A1C1\\">;</span></span></code></pre>"`;
